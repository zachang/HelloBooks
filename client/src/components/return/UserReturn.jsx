import React from 'react';
import PropTypes from 'react-proptypes';

const UserReturn = props => (
  <div className='col l4 s10 m6 cardsm'>
    <div className='card large sticky-action'>
      <div className='card-image waves-effect waves-block waves-light'>
        <img className='activator'
             src={(props.returning.Book.book_image !== null) ? props.returning.Book.book_image : '../../imgs/default.jpg'}/>
      </div>
      <div className='card-content'>
        <span className='card-title activator grey-text text-darken-4'>
          <small>{ props.returning.Book.book_name }</small>
          <i className='material-icons right'>more_vert</i>
        </span>
      </div>
      <div className='card-reveal'>
        <span className='card-title grey-text text-darken-4'>
          <small>{ props.returning.Book.book_name }</small>
          <i className='material-icons right'>
            close
          </i>
        </span>
        <p>Author: { props.returning.Book.author }</p>
        <p>Category: { props.returning.Book.Category.category_name }</p>
        <p>Published: { new Date(props.returning.Book.publish_year).getFullYear() }</p>
        <p>Pages: { props.returning.Book.pages }</p>
        <p>Stock: { props.returning.Book.book_count }</p>
        <p>Description: {
          (props.returning.Book.description)
        }
        </p>
      </div>
      <div className='card-action home-card'>
      </div>
    </div>
  </div>
);

UserReturn.propTypes = {
  returning: PropTypes.object.isRequired,

};
export default UserReturn;
