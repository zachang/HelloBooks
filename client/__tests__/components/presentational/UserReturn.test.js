import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import UserReturn from '../../../src/components/return/UserReturn.jsx';

describe('<UserReturn/>', () => {
  const props = {
    returning: {
      Book: {
        bookName: 'Science',
        bookCount: 4,
        pages: 600,
        description: 'fine',
        publishYear: new Date(),
        id: 1,
        Category: {
          categoryName: 'Arts'
        }
      },
      User: {
        fullname: 'Tito Navo'
      }
    }
  };

  const shallowWrapper = shallow(<UserReturn {...props}/>);
  const tree = toJson(shallowWrapper);


  it('renders <UserReturn/> component', () => {
    expect(tree).toMatchSnapshot();
  });
});

