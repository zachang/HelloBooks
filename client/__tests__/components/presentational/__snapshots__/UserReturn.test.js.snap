// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserReturn/> renders <UserReturn/> component 1`] = `
<div
  className="col l4 s10 m6 cardsm"
>
  <div
    className="card large sticky-action"
  >
    <div
      className="card-image waves-effect waves-block waves-light"
    >
      <img
        className="activator"
        src="../../imgs/default.jpg"
      />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title activator grey-text text-darken-4"
      >
        <small>
          Science
        </small>
        <i
          className="material-icons right"
        >
          more_vert
        </i>
      </span>
    </div>
    <div
      className="card-reveal"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        <small>
          Science
        </small>
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
      <p>
        Author: 
      </p>
      <p>
        Category: 
        Arts
      </p>
      <p>
        Published: 
        2018
      </p>
      <p>
        Pages: 
        600
      </p>
      <p>
        Stock: 
        4
      </p>
      <p>
        Description: 
        fine
      </p>
    </div>
    <div
      className="card-action home-card"
    />
  </div>
</div>
`;
